infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -144,11 +144,15 @@ public IClientRouter addClient(IClient client) {
         client.setRouter(this);
 
         // Iterate through all types of CorfuMsgType, registering the handler
-        client.getHandledTypes().stream()
-                .forEach(x -> {
-                    handlerMap.put(x, client);
-                    log.info("Registered client to handle messages of type {}", x);
-                });
+        try {
+            client.getHandledTypes().stream()
+                    .forEach(x -> {
+                        handlerMap.put(x, client);
+                        log.info("Registered client to handle messages of type {}", x);
+                    });
+        } catch (Exception ex) {
+            // No Legacy getHandledTypes - Ignore
+        }
```

Comment:
Is the exception expected to be thrown when getHandledTypes() returns an empty set? Will it be better to explicitly check if it's empty and log the warning/info? I thought it won't throw a NPE if the streaming collection is empty.